services/brig/src/Brig/Provider/DB.hs

```diff
@@ -274,12 +274,14 @@ deleteService :: MonadClient m
     -> Name
     -> RangedServiceTags
     -> m ()
-deleteService pid sid name tags =  retry x5 $ batch $ do
-    setConsistency Quorum
-    setType BatchUnLogged
-    addPrepQuery cql (pid, sid)
-    deleteServicePrefix sid name
-    deleteServiceTags pid sid name tags
+deleteService pid sid name tags = do
+    retry x5 $ batch $ do
+        setConsistency Quorum
+        setType BatchUnLogged
+        addPrepQuery cql (pid, sid)
+        deleteServicePrefix sid name
+        deleteServiceTags pid sid name tags
+    deleteServiceWhitelist Nothing pid sid
```

Comment:
`deleteServiceWhitelist` is done as a separate step because it's a more complicated operation (you have to consult a reverse index and delete the service from all returned teams' whitelists).

Reply:
As mentioned `deleteServiceWhitelist` is a complex operation and it grows linearly in the number of teams so it's possible that it times out; if that happens, and since `deleteServiceWhitelist` is not done in a transaction, we would end up with a deleted service which is still part of a handful of teams' whitelist so we should either:

1. Call `deleteServiceWhitelist` first; if this fails, it's no problem to simply repeat it
2. Gather all teams first, and then delete the service.
3. Do all this asynchronously

`3.` could (should) be done together with one of the other 2 options.